cmd/aws-vpc-cni-init/main.go

```diff
@@ -0,0 +1,177 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// The aws-node initialization
+package main
+
+import (
+	"os"
+
+	"github.com/aws/amazon-vpc-cni-k8s/utils/cp"
+	"github.com/aws/amazon-vpc-cni-k8s/utils/imds"
+	"github.com/aws/amazon-vpc-cni-k8s/utils/sysctl"
+	log "github.com/sirupsen/logrus"
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	defaultHostCNIBinPath = "/host/opt/cni/bin"
+	vpcCniInitDonePath    = "/vpc-cni-init/done"
+	metadataLocalIP       = "local-ipv4"
+	metadataMAC           = "mac"
+
+	envDisableIPv4TcpEarlyDemux = "DISABLE_TCP_EARLY_DEMUX"
+	envEnableIPv6               = "ENABLE_IPv6"
+	envHostCniBinPath           = "HOST_CNI_BIN_PATH"
+)
+
+func getEnv(env, def string) string {
+	if val, ok := os.LookupEnv(env); ok {
+		return val
+	}
+	return def
+}
+
+func main() {
+	os.Exit(_main())
+}
+
+func _main() int {
+	log.Debug("Started Initialization")
+	pluginBins := []string{"loopback", "portmap", "bandwidth", "aws-cni-support.sh"}
```

Comment:
We need `host-local` plugin for IPv6  - https://github.com/aws/amazon-vpc-cni-k8s/blob/master/misc/10-aws.conflist#L23

Reply:
I do see `host-local` in entrypoint script here: https://github.com/aws/amazon-vpc-cni-k8s/blob/master/scripts/entrypoint.sh#L154 , but it is not present in init script: https://github.com/aws/amazon-vpc-cni-k8s/blob/master/scripts/init.sh#L25 . I assume it should be in both?

Also, it does seem like IPv6 test suite passes without this. I will add it, though.

Reply:
I think we can keep it only in init script. It is confusing to have it in two places. We can take this chance to cleanup the code.

Reply:
btw @achevuru , some of my code was implemented before v6 hence it is a good callout to test v6 before merging.

Reply:
but when someone sets `AWS_VPC_K8S_CNI_CONFIGURE_RPFILTER` to true, which is the default, then it is main container that should install this binary

Reply:
Init container will always be called right? so one place should be fine. I think it was added initially when init container was first supported..

Reply:
I guess the question then is whether we even support `AWS_VPC_K8S_CNI_CONFIGURE_RPFILTER` being configured as `"false"`. Currently, both the init container and the main container installs these binaries. Since the init container does it unconditionally, there is no need for the main container to do it either way

Reply:
nvm, it looks like `AWS_VPC_K8S_CNI_CONFIGURE_RPFILTER` is used here: https://github.com/aws/amazon-vpc-cni-k8s/blob/f21b587fdf6370a01b181ddce797398eb7951a23/pkg/networkutils/network.go#L298 . So really, this env variable does not make sense as a gate for the main entrypoint installing the same binaries that init entrypoint already did

Reply:
Ran all IPv6 and IPv4 test suites and verified that they passed. Modified entrypoint scripts such that only init container installs external plugins. This also allowed me to simplify dockerfiles.